src/components/treeActivity/index.tsx

```diff
@@ -35,19 +35,146 @@ const EntryMessage = styled(Paragraph)`
   color: ${TEXT_GREY};
 `;
 
+const StewardshipActivityDropdownContainer = styled.div`
+  margin: -10px 0 -5px 5px;
+`;
+
+const StewardshipActivityDropdown = styled.div`
+  font-size: 15px;
+  color: ${MID_GREEN};
+`;
+
+const CenteredPagination = styled(Pagination)`
+  display: flex;
+  justify-content: center;
+  & .ant-pagination-item-active {
+    background-color: white;
+  }
+  & .ant-pagination-item-active :hover {
+    background-color: ${DARK_GREEN};
+  }
+`;
+
 interface TreeActivityProps {
   readonly stewardship: TreeCare[];
-  readonly limit?: number;
 }
 
-const TreeActivity: React.FC<TreeActivityProps> = ({ stewardship, limit }) => {
+const TreeActivity: React.FC<TreeActivityProps> = ({ stewardship }) => {
+  const months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  // convert a month from its three letter abbreviation (e.g. 'Jan') to its long form (e.g. 'January')
+  const monthShortToLong = (shortMonth: string): string => {
+    for (const month of months) {
+      if (shortMonth.substring(0, 3) === month.substring(0, 3)) {
+        return month;
+      }
+    }
+    return '';
+  };
+
+  // sort two dates by a combination of year and month
+  const sortByYearMonth = (
+    yearMonth1: { label: string },
+    yearMonth2: { label: string },
+  ) => {
+    const label1 = yearMonth1.label;
+    const label2 = yearMonth2.label;
+
+    const year1 = label1.slice(-4);
+    const month1 = months.indexOf(label1.split(' ')[0]);
+    const year2 = label2.slice(-4);
+    const month2 = months.indexOf(label2.split(' ')[0]);
+
+    if (year1 === year2 && month1 === month2) {
+      return 0;
+    } else if (year1 < year2 || (year1 === year2 && month1 < month2)) {
+      return -1;
+    } else {
+      return 1;
+    }
+  };
+
+  const [selectedMonthYear, setSelectedMonthYear] = useState(
+    new Date().toLocaleString('default', {
+      month: 'short',
+      year: 'numeric',
+    }),
+  );
+  const selectedMonthYearStewardship: TreeCare[] = stewardship.filter(
+    (entry) =>
+      // choose stewardship activities with the selected month and year
+      entry.date.substring(0, 3) === selectedMonthYear.substring(0, 3) &&
+      entry.year === parseInt(selectedMonthYear.slice(-4), 10),
+  );
+  const monthYearOptions = stewardship
+    // remove activities with duplicate year and date
+    .filter(
+      (entry, index, self) =>
+        index ===
+        self.findIndex(
+          (e) =>
+            e.year === entry.year &&
+            e.date.substring(0, 3) === entry.date.substring(0, 3),
+        ),
+    )
+    // turn the filtered array into an array of labels and values
+    .map((entry) => {
+      return {
+        label: monthShortToLong(entry.date) + ' ' + entry.year,
+        value: entry.date.substring(0, 3) + ' ' + entry.year,
```

Comment:
💭 You could have `value` be of type `{month: string, year: number}` instead of just `string` to avoid having to extract the month vs year everywhere with `substring` and `splice`

Reply:
I tried doing this but couldn't because the `options` attribute of the `Select` component wants something of type `(OptionData | OptionGroupData)[]` and `{ label: string, value: { month: string, year: number } }` doesn't match that type. Specifically, it's not an `OptionData` because the `value` field is not of type `Key` (which is an alias for `string | number`). But it's also not an `OptionGroupData` because it doesn't have the `options` field. Should I leave the `value` field as is and add a third field of type `{ month: string, year: number }` or is there a better way to do it?

![image](https://user-images.githubusercontent.com/33704204/140812340-f7e11a2b-3d9c-472b-8404-26c4e379b66d.png)

![image](https://user-images.githubusercontent.com/33704204/140811913-da83cc02-c4c0-4654-a81b-069a7f2d5c3a.png)